dali/operators/decoder/nvjpeg/decoupled_api/nvjpeg_memory.cc

```diff
@@ -208,17 +208,29 @@ static int DeviceNew(void **ptr, size_t size) {
     *ptr = nullptr;
     return cudaSuccess;
   }
-  *ptr = GetBuffer(std::this_thread::get_id(), AllocType::GPU, size);
-  return *ptr != nullptr ? cudaSuccess : cudaErrorMemoryAllocation;
+  // this function should not throw, but return a proper result
+  try {
+    *ptr = GetBuffer(std::this_thread::get_id(), AllocType::GPU, size);
+    return *ptr != nullptr ? cudaSuccess : cudaErrorMemoryAllocation;
+  } catch (...) {
+    *ptr = nullptr;
+    return cudaErrorMemoryAllocation;
```

Comment:
I don't think we should at least try to return a meaningful status
```suggestion
  } catch (const std::bad_alloc &) {
    *ptr = nullptr;
    return cudaErrorMemoryAllocation;
  } catch (const CUDAError &e) {
    *ptr = nullptr;
    return e.is_rt_api() ? e.rt_error() : cudaErrorUnknown;
  } catch (const std::exception &e) {
    // print? should we catch it at all?
    *ptr = nullptr;
    return cudaErrorUnknown;
  } catch (...) {
    // Now that's really heavy stuff - print something?
    // Should we really catch it?
    *ptr = nullptr;
    return cudaErrorUnknown;
  }
```

Reply:
Done